cmd/fleetctl/mdm.go

```diff
@@ -52,55 +53,105 @@ func mdmRunCommand() *cli.Command {
+			// fetch all specified hosts by their identifier
+			//
+			// Note that this retrieves all hosts sequentially as it assumes the
+			// number of hosts will typically be small. If we need to improve on
+			// this, we could use something like the errgroup package to retrieve
+			// them concurrently or better yet add a batch endpoint to get hosts by
+			// the loose "identifier".
+			var (
+				hostUUIDs     []string
+				notFoundCount int
+				platform      string
+			)
+			for _, ident := range hostIdents {
+				host, err := client.HostByIdentifier(ident)
+				if err != nil {
+					var nfe service.NotFoundErr
+					if errors.As(err, &nfe) {
+						notFoundCount++
+						continue
 					}
+
+					var sce kithttp.StatusCoder
+					if errors.As(err, &sce) {
+						if sce.StatusCode() == http.StatusForbidden {
+							return fmt.Errorf("You don't have permission to run an MDM command on one or more specified hosts: %w", err)
+						}
+					}
+					return err
 				}
-				return err
+
+				if host.Platform != platform && platform != "" {
+					return errors.New(`Command can't run on hosts with different platforms. Make sure the hosts specified in the "hosts" flag are either all macOS or all Windows hosts.`)
+				}
+				platform = host.Platform
+
+				// TODO(mna): this "On" check is brittle, but looks like it's the only
+				// enrollment indication we have right now...
+				if host.MDM.EnrollmentStatus == nil || !strings.HasPrefix(*host.MDM.EnrollmentStatus, "On") ||
+					host.MDM.Name != fleet.WellKnownMDMFleet {
+					return errors.New(`Can't run the MDM command because one or more hosts have MDM turned off. Run the following command to see a list of hosts with MDM on: fleetctl get hosts --mdm.`)
+				}
+
+				hostUUIDs = append(hostUUIDs, host.UUID)
 			}
 
-			// TODO(mna): this "On" check is brittle, but looks like it's the only
-			// enrollment indication we have right now...
-			if host.MDM.EnrollmentStatus == nil || !strings.HasPrefix(*host.MDM.EnrollmentStatus, "On") ||
-				host.MDM.Name != fleet.WellKnownMDMFleet {
-				return errors.New("Can't run the MDM command because the host doesn't have MDM turned on. Run the following command to see a list of hosts with MDM on: fleetctl get hosts --mdm")
+			if len(hostUUIDs) == 0 {
+				// all hosts were not found
+				return errors.New("No hosts targeted. Make sure you provide a valid hostname, UUID, osquery host ID, or node key.")
+			}
+			if notFoundCount > 0 {
+				// at least one was not found
+				return errors.New("One or more targeted hosts don't exist. Make sure you provide a valid hostname, UUID, osquery host ID, or node key.")
 			}
 
-			result, err := client.EnqueueCommand([]string{host.UUID}, payload)
+			result, err := client.RunMDMCommand(hostUUIDs, payload, platform)
 			if err != nil {
+				if errors.Is(err, service.ErrMissingLicense) && platform == "windows" {
+					return errors.New("Missing or invalid license. Wipe command is available in Fleet Premium only.")
```

Comment:
Currently that's the only command on Windows that requires premium, but I'm also not a fan of messages like that because it changes at a distance - if we introduce a new premium command, then that message would be outdated/incomplete. 

Writing this comment, I think what would help a bit would be to add the message string as a constant near the validation, so that if that validation changes, it's easy(ier) to see that the message should change too. I'll make that change.